yaml/xpu_functions.yaml

```diff
@@ -55,3 +55,27 @@ supported:
   - ge.Tensor
   - ge.Tensor_out
   - ge_.Tensor
+  - abs
+  - abs_
+  - abs.out
+  - sin
+  - sin_
+  - sin.out
+  - cos
+  - cos_
+  - cos.out
+  - log
+  - log_
+  - log.out
+  - sqrt
+  - sqrt_
+  - sqrt.out
+  - rsqrt
+  - rsqrt_
+  - rsqrt.out
+  - tanh
+  - tanh_
+  - tanh.out
+  - neg
+  - neg_
+  - neg.out
```

Comment:
Why do we need the yaml file? For code generation?

Reply:
yes, for codegen. yaml file helps to generate their wrapper to register op to stock PyTorch.

Reply:
1. To save efforts and lines of code in sources.
2. Help us to verify latest ATen prototype in build time.

We don't need such codes any more,
![image](https://github.com/intel/torch-xpu-ops/assets/28250760/a8d645ec-83a3-48e8-959a-b5d0ff384369)

Reply:
Since we just have some limited operations, I do **NOT** think code generation is a necessary component to be maintained for the interim goal. Meanwhile, how could we ensure the code generation logic is align with CUDA?

Reply:
Regarding these codes, we need these codes for in-tree. We are at the first stage of in-tree. So everything should follow the in-tree standard.

Reply:
1. `limited` is not that true. We will almost have half numbers (360) of operators compared with IPEX by the end of March. And by the end of PyTorch2.5 release, we might have more.
2. There are several ATen registration approaches in-tree. CPU/CUDA like, vulkan like, and lazy tensor like. It is simply migrate between these three approach. And I don't think it will impact in-tree things.

So the purpose here is for torch-xpu-ops development.

Reply:
If we have a risk to support 360 operations, we can have more discussions. It is another separate topic.

But for the upstreaming, it is not worth us to do such development for an interim goal. For now, I even cannot quite understand the code directory structure is conflict with PyTorch design.

For example, all the device-specific operations are under `ATen/native/${device}/` in the stock PyTorch. 

Regarding the registration mechanism, the rule should be :

- If the stub is there, we should use `REGISTER_DISPATCH` to register SYCL kernel - https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/mps/operations/BinaryKernel.mm#L371
- If the stub is not there, we can use `TORCH_IMPL_FUNC`